### ISSUE TITLE
Property type errors in an `as const` literal should be located at the property

### ISSUE THREAD

**Issue by Peeja** [microsoft/TypeScript#54374]
# Bug Report

### 🔎 Search Terms

as const, property

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about `as const`

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgGIHt3IN4ChnJwBcyAjANz7IBGJATJQL664D0ryA6gBYQjIh0YZAAM4AZ2QJ0IcWBEAaZGF7JoUdFGTBJAG3QI4kACaFhKlNTimADhpvQwATwB0uabOExMAWQCuYHDUuhAkGFgAvDhUxGQKVLTIAMy4zGwcAEIByADuvPxikh5yisqq6praklBwOhCmfjbKWBbINnBQfMLGEAi6HUbAMm7FXpgAwjJyYZjIUXgEsaTxBIkpjIRFU2BAA)

### 💻 Code

```ts
interface Foo {
  a: 1;
  b: 2;
}

// When not `as const`, the error is located at the bad property.
const fooMutable: Foo = {
  a: 1,
  b: 3 // ❌
}

// But when `as const`, the error is raised up to the parent declaration.
const fooConst: Foo = { // ❌
  a: 1,
  b: 3
} as const
```

### 🙁 Actual behavior

`fooConst` is underlined in red, with error:

```
Type '{ readonly a: 1; readonly b: 3; }' is not assignable to type 'Foo'.
  Types of property 'b' are incompatible.
    Type '3' is not assignable to type '2'.(2322)
```

### 🙂 Expected behavior

`fooConst`'s `b` should be underlined in red, with the same error as `fooMutable`'s `b`:

```
Type '3' is not assignable to type '2'.(2322)
input.tsx(3, 3): The expected type comes from property 'b' which is declared here on type 'Foo'
```

---

For large object literals which use `as const`, this makes for a more difficult experience when one property is incorrect.

**Comment by DanielRosenwasser** [comment]
Thanks @Andarist!
